packages/dds/sequence/src/test/endpointInRangeIndex.spec.ts

```diff
@@ -0,0 +1,186 @@
+		// sort the query result by the interval endpoint value
+		compareFn = (a: Interval, b: Interval) => {
+			if (a.end === b.end) {
+				return a.start - b.start;
+			}
+			return a.end - b.end;
+		};
+	});
+
+	it("return empty result when the index is empty", () => {
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(1, 1);
+		assert.equal(results.length, 0, "Should not return anything once the index is empty");
+	});
+
+	it("limit on the start/end value", () => {
+		endpointInRangeIndex.add(createTestInterval(1, 1));
+		endpointInRangeIndex.add(createTestInterval(1, 3));
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(2, 1);
+		assert.equal(results.length, 0, "The start value should not be lower than the end value");
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(0, 1);
+		assert.equal(results.length, 0, "The start value should be larger than 0");
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(-2, -1);
+		assert.equal(results.length, 0, "The endpoint value should not be negative");
+	});
+
+	it("can find correct results after adding multiple intervals", () => {
+		endpointInRangeIndex.add(createTestInterval(1, 1));
+		endpointInRangeIndex.add(createTestInterval(1, 3));
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(1, 1);
+		assertPlainNumberIntervals(results, [{ start: 1, end: 1 }]);
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(1, 3);
+		assertPlainNumberIntervals(results, [
+			{ start: 1, end: 1 },
+			{ start: 1, end: 3 },
+		]);
+
+		endpointInRangeIndex.add(createTestInterval(2, 4));
+		endpointInRangeIndex.add(createTestInterval(3, 5));
+		endpointInRangeIndex.add(createTestInterval(3, 4));
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(3, 5);
+		results.sort(compareFn);
+		assertPlainNumberIntervals(results, [
+			{ start: 1, end: 3 },
+			{ start: 2, end: 4 },
+			{ start: 3, end: 4 },
+			{ start: 3, end: 5 },
+		]);
+	});
+
+	it("can find correct results after removing multiple intervals", () => {
+		const interval1 = createTestInterval(1, 1);
+		const interval2 = createTestInterval(1, 3);
+		endpointInRangeIndex.add(interval1);
+		endpointInRangeIndex.add(interval2);
+		endpointInRangeIndex.remove(interval1);
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(1, 2);
+		assert.equal(results.length, 0);
+
+		const interval3 = createTestInterval(2, 4);
+		endpointInRangeIndex.add(interval3);
+		endpointInRangeIndex.remove(interval2);
+
+		results = endpointInRangeIndex.findIntervalsWithEndpointInRange(3, 5);
+		assertPlainNumberIntervals(results, [{ start: 2, end: 4 }]);
+	});
+
+	it("compare with simple query method using random inputs", () => {
+		const testIndex = new TestEndpointInRangeIndex();
+
+		// Generate intervals randomly and add them to both index
+		const intervals = generateRandomIntervals(10, 1, 20);
```

Comment:
nit: in test code, always prefer named parameter objects for helper functions like this. It's a whole lot easier to read `generateRandomIntervals({ count: 10, min: 1, max: 20 })` than the 3 magic numbers here.

Reply:
i'd probably ramp this test up to 100 or 1000 as well, btw. It should still run significantly faster than our fuzz tests.